frontend/src/pages/modelServing/screens/global/utils.ts

```diff
@@ -11,10 +12,15 @@ export const getTokenDisplayName = (secret: SecretKind): string =>
 
 export const getInferenceServiceActiveModelState = (
   is: InferenceServiceKind,
-): InferenceServiceModelState =>
-  <InferenceServiceModelState | undefined>is.status?.modelStatus.states.activeModelState ||
-  <InferenceServiceModelState | undefined>is.status?.modelStatus.states.targetModelState ||
-  InferenceServiceModelState.UNKNOWN;
+): InferenceServiceModelState => {
+  if (typeof is.status?.modelStatus.states.activeModelState !== 'undefined') {
+    return asEnumMember(is.status.modelStatus.states.activeModelState, InferenceServiceModelState)!;
+  }
+  if (typeof is.status?.modelStatus.states.targetModelState !== 'undefined') {
+    return asEnumMember(is.status.modelStatus.states.targetModelState, InferenceServiceModelState)!;
+  }
+  return InferenceServiceModelState.UNKNOWN;
+};
```

Comment:
Update the enum functions to support `undefined` as an input.
```suggestion
): InferenceServiceModelState =>
  asEnumMember(is.status?.modelStatus.states.activeModelState, InferenceServiceModelState) ||
  asEnumMember(is.status?.modelStatus.states.targetModelState, InferenceServiceModelState) ||
  InferenceServiceModelState.UNKNOWN;
```